docs/DeveloperGuide.md

```diff
@@ -238,33 +238,84 @@ _{more aspects and alternatives to be added}_
+* `Model#getFIlteredPersonList()`
+* `Model#addPerson(Person)`
+* `Model#storePreviousUndoableCommand(String)`
+* `Model#resetRedoableStateList()`
+* `Model#resetUndoableStateList()`
+* `Model#removeRedoCommands()`
 
 Given below is an example usage scenario and how the clone mechanism behaves at each step.
 
-Step 1. The user executes "list" to see what Persons are available in the address book
+Step 1. The user launches the application for the first time. 
+
+Step 2. The user executes `list` to see what Persons are available in the address book
 
 ![Clone0](images/Clone0.png)
 
-Step 2. The user executes "clone 1" to clone the person at index 1 of the addressbook, John
+Step 3. The user executes `clone 1` to clone the person at index 1 of the address book, John. The `clone` command first
+calls `CloneCommand#execute`, which in turn checks to see if the index provided is valid in CloneCommandParser by
+calling `ParserUtil#parseIndex`. If the index is valid, `Index#getZeroBased` is then called to ensure that it is
+smaller than the size of the list. In this case, the index 1 provided is valid.
 
-![Clone1](images/Clone1.png)
+![CloneActivityDiagram0](images/CloneActivityDiagram0.png)
 
-The following activity diagram summarizes what happens when a user executes a new command:
+After all the checks on the index have been done, cloneCommand then calls `CloneCommand#clonePerson`, which separates
+the Person name (as a string) into two substrings using `CloneCommand#splitStringAtLastSpace`, the name and the possible
+suffix, respectfully.
+
+After being separated into two substrings, `CloneCommand#clonePerson` then separates the person name into three cases:
+Firstly, if the name has no spaces and no suffix. Secondly, if the name has spaces but the suffix does not consist of
+only an integer. And lastly, if the name has spaces and the suffix consists of only an integer. As John has no spaces
+and no suffix, this would be the first case highlighted.
+
+Based on these three cases, either than have a suffix or they do not. If they do not have a suffix, such as in the first
+two cases, then they have a suffix of one added to the back of the person name. If they name already possesses a suffix,
+then this suffix is incremented. The newly cloned person with their new name is now returned to `CloneCommand#execute`.
+As John has no suffix, a suffix of one is added to the back of his name and John 1 is returned. 
+
+![CloneActivityDiagram1](images/CloneActivityDiagram1.png)
+
+Once returned, the command then attempts to add the cloned person into the list through `Model#addPerson`. However, if
+it throws a duplicate exception (implying that a person with that name already exists in the contact book), then this
+cloned person is cloned yet again via `CloneCommand#clonePerson` until the suffix of the cloned person returned is
+unique. As the only other person in the list prior was James, John 1 is added to the list successfully on the first
```

Comment:
```suggestion
unique. As the only other person in the list prior was John, John 1 is added to the list successfully on the first
```

Not entirely sure I am interpreting this part correctly, but maybe this is a typo?